EIPS/eip-1605.md

```diff
@@ -0,0 +1,404 @@
+---
+eip: 1605
+title: Contract ABI encoding specification
+author: Christian Reitwiessner (@chriseth), Gavin Wood, Alex Beregszaszi (@axic) et al.
+type: Standards Track
+category: Interface
+status: Draft
+created: 2018-10-20
+---
+
+## Abstract
+
+The Contract Application Binary Interface (ABI) is the standard way to interact with contracts in the Ethereum ecosystem, both from outside the blockchain and for contract-to-contract interaction.
+
+## Specification
+
+### Basic Design
+
+Data is encoded according to its type,
+as described in this specification.  The encoding is not self describing and thus requires a schema in order to decode.
+
+We assume the interface functions of a contract are strongly typed, known at compilation time and static. We assume that all contracts will have the interface definitions of any contracts they call available at compile-time.
+
+This specification does not address contracts whose interface is dynamic or otherwise known only at run-time.
+
+### Function Selector
+
+The first four bytes of the call data for a function call specifies the function to be called. It is the
+first (left, high-order in big-endian) four bytes of the Keccak-256 (SHA-3) hash of the signature of the function. The signature is defined as the canonical expression of the basic prototype without data location specifier, i.e.
+the function name with the parenthesised list of parameter types. Parameter types are split by a single comma - no spaces are used.
+
+Note:
+    The return type of a function is not part of this signature. In Solidity's function overloading return types are not considered. The reason is to keep function call resolution context-independent.
+
+### Argument Encoding
+
+Starting from the fifth byte, the encoded arguments follow. This encoding is also used in other places, e.g. the return values and also event arguments are encoded in the same way, without the four bytes specifying the function.
+
+### Types
+
+The following elementary types exist:
+
+- ``uint<M>``: unsigned integer type of ``M`` bits, ``0 < M <= 256``, ``M % 8 == 0``. e.g. ``uint32``, ``uint8``, ``uint256``.
```

Comment:
I don't know that
```
``apple``
```
is standard markdown for inline code snippets.  Usually it is
```
`apple`
```
Recommend changing to single ` escapes even if it will render correctly in Jekyll just because it is less noisy in the raw form.